resolver/resolver.go

```diff
@@ -257,6 +257,15 @@ type Target struct {
 	URL url.URL
 }
 
+// ToEndpoint retrieves endpoint from either URL.Path or URL.Opaque.
```

Comment:
Please also document that we strip the leading `/`.

And in `parseTarget()`, we strip the leading `/` even when we are using `URL.Opaque`. At this point, I don't even know if that is the correct thing to do. But it was done to ensure that there was no behavior change (when we switched our target parsing to be more in tune with RFC 3986). See https://github.com/grpc/grpc-go/blob/master/clientconn.go#L1609-L1613

We should keep the same behavior here as well.

Reply:
Done. 

> At this point, I don't even know if that is the correct thing to do.  But it was done to ensure that there was no behavior change (when we switched our target parsing to be more in tune with RFC 3986).

Out of curiosity, is standardizing the approach across all resolver implementations - particularly unix - the challenge?

Reply:
The Unix resolver is a little special because of the way it was first implemented and we cannot completely change the implementation to standardize it with other implementations. I probably mentioned in some other comment, but we did undertake a reasonably sized effort to standardize the way we parse the target strings and we also cleaned up the unix resolver the best we could. Any more standardization would lead to change in existing behavior, which could break current users.